scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala

```diff
@@ -67,14 +67,23 @@ class ScioContextTest extends PipelineSpec {
     pipeline.getOptions.getTempLocation shouldBe expected
   }
 
-  it should "support user defined job name" in {
+  it should "support user defined job name via options" in {
     val jobName = "test-job-1"
     val opts = PipelineOptionsFactory.create().as(classOf[DataflowPipelineOptions])
     opts.setJobName(jobName)
     val pipelineOpts = ScioContext(opts).pipeline.getOptions.as(classOf[DataflowPipelineOptions])
     pipelineOpts.getJobName shouldBe jobName
   }
 
+  it should "support user defined job name via context" in {
+    val jobName = "test-job-1"
+    val opts = PipelineOptionsFactory.create().as(classOf[DataflowPipelineOptions])
+    val sc = ScioContext(opts)
+    sc.setJobName(jobName)
+    val pipelineOpts = sc.pipeline.getOptions.as(classOf[DataflowPipelineOptions])
+    pipelineOpts.getJobName shouldBe jobName
+  }
+
```

Comment:
Could you please also add test where:
 * job name is set via options
 * followed by set (with different name) via context
and the end name should be the one provided via  context.